src/widgets/tables/merging_sets_screen.py

```diff
@@ -0,0 +1,409 @@
+        except ValueError as e:
+            self.import_state_label.setText(str(e))
+        else:
+            self._on_success(reader)
+
+    def _on_success(self, reader, file_path=None):
+        if file_path is not None:
+            self.filepath_line.setText(file_path)
+        self.import_state_label.clear()
+        if self.engine.is_data_big():
+            error = QMessageBox()
+            error.setIcon(QMessageBox.Warning)
+            error.setText("This file is too big.\nYou must save it in database!")
+            error.setWindowTitle("Warning")
+            error.exec_()
+        self.new_data = reader.read(None)
+        last_preview = self.layout.takeAt(2).widget()
+        if last_preview is not None:
+            last_preview.deleteLater()
+        self._render_table(self.new_data_view, False)
+
+        for i in reversed(range(self.columns_left_layout.count() - 1)):
+            # workaround to delete as low content as possible, removing nested layouts was giving me some segfault
+            self.columns_left_layout.itemAt(i).widget().setParent(None)
+
+        for i in reversed(range(self.columns_right_layout.count() - 1)):
+            self.columns_right_layout.itemAt(i).widget().setParent(None)
+
+        imported_columns = self._get_imported_columns()
+        for column in imported_columns:
+            self.columns_left_layout.insertWidget(
+                self.columns_left_layout.count() - 1,
+                ColumnWidget(column, self.columns_left_layout),
+            )
+
+        if self.new_data is not None and self.new_data.columns is not None:
+            for column in sorted(self.new_data.columns, key=lambda x: x.upper()):
+                self.columns_right_layout.insertWidget(
+                    self.columns_right_layout.count() - 1,
+                    ColumnWidget(column, self.columns_right_layout),
+                )
+
+        self.submit_button.setEnabled(self.new_data is not None)
+
+    def _on_submit(self):
+        new_columns_left = []
+        new_columns_right = []
+        for i in range(self.columns_left_layout.count() - 1):
+            new_columns_left.append(
+                self.columns_left_layout.itemAt(i).widget().column_name
+            )
+        for i in range(self.columns_right_layout.count() - 1):
+            new_columns_right.append(
+                self.columns_right_layout.itemAt(i).widget().column_name
+            )
+
+        if len(new_columns_left) != len(new_columns_right):
+            self._render_equality_warning()
+            if self.cancel_merge:
+                self.cancel_merge = False
+                return
+
+        num_columns = max(len(new_columns_left), len(new_columns_right))
+        labels_left = new_columns_left[:num_columns]
+        labels_right = new_columns_right[:num_columns]
+        for i in range(len(new_columns_right) - len(new_columns_left)):
+            new_column_name = f"Column {i}"
```

Comment:
when there are more columns on the right than on the left, the columns have these generic names after merging. can't we instead use the names of the actual columns from the right (`new_columns_right[i+...])`? we could add some sort of unique identifier to the end to make it more likely to be unique if that's the problem